internal/k8s/reconciler/http_route.go

```diff
@@ -163,6 +164,18 @@ func convertHTTPRouteFilters(routeFilters []gwv1alpha2.HTTPRouteFilter) []core.H
 					Remove: filter.RequestHeaderModifier.Remove,
 				},
 			})
+		case gwv1alpha2.HTTPRouteFilterURLRewrite:
+			if filter.URLRewrite.Path != nil {
+				if filter.URLRewrite.Path.ReplacePrefixMatch != nil {
+					filters = append(filters, core.HTTPFilter{
+						Type: core.HTTPURLRewriteFilterType,
+						URLRewrite: core.HTTPURLRewriteFilter{
+							Type:               core.ReplacePrefixMatchURLRewriteType,
+							ReplacePrefixMatch: *filter.URLRewrite.Path.ReplacePrefixMatch,
+						},
+					})
+				}
+			}
```

Comment:
I believe we need to check `filter.URLRewrite.Path.Type` as well; otherwise, you could specify something like this and get a prefix rewrite when you technically indicated a full path rewrite:
```yaml
filters:
- type: URLRewrite
  urlRewrite:
    path:
      type: ReplaceFullPath
      replacePrefixMatch: /foo
      replaceFullPath: /bar
```

Reply:
The thing I like about this structure is that early returns for invalid cases result in less conditional nesting ❤️ 
```suggestion
		case gwv1alpha2.HTTPRouteFilterURLRewrite:
			// We currently only support prefix match replacement
			if filter.URLRewrite.Path == nil ||
				filter.URLRewrite.Path.Type != gwv1alpha2.PrefixMatchHTTPPathModifier ||
				filter.URLRewrite.Path.ReplacePrefixMatch == nil {
				continue
			}

			filters = append(filters, core.HTTPFilter{
				Type: core.HTTPURLRewriteFilterType,
				URLRewrite: core.HTTPURLRewriteFilter{
					Type:               core.ReplacePrefixMatchURLRewriteType,
					ReplacePrefixMatch: *filter.URLRewrite.Path.ReplacePrefixMatch,
				},
			})
```